### PR TITLE
adding a Dataflow server local

### PR DESCRIPTION
**Pull request by joshlong**
adds the `org.springframework.cloud:spring-cloud-starter-dataflow-server-local` dependency to a generated build from the Spring Initializr

### CODE REVIEW

File: initializr-service/application.yml [hunk 1]
```diff
@@ -441,6 +441,16 @@ initializr:
           description: Messaging on AWS with SQS and spring-cloud-aws-messaging
           groupId: org.springframework.cloud
           artifactId: spring-cloud-starter-aws-messaging
+    - name: Cloud Dataflow
```
Comment (markfisher) [inline initializr-service/application.yml]: should be 2 words: Data Flow

File: initializr-service/application.yml [hunk 2]
```diff
@@ -441,6 +441,16 @@ initializr:
...
+      bom: cloud-bom
+      versionRange: "1.1.0.BUILD-SNAPSHOT"
+      content:
+        - name: Local Dataflow Server
```
Comment (markfisher) [inline initializr-service/application.yml]: 2 words: Data Flow

File: initializr-service/application.yml [hunk 3]
```diff
@@ -441,6 +441,16 @@ initializr:
           description: Messaging on AWS with SQS and spring-cloud-aws-messaging
           groupId: org.springframework.cloud
           artifactId: spring-cloud-starter-aws-messaging
+    - name: Cloud Dataflow
+      bom: cloud-bom
+      versionRange: "1.1.0.BUILD-SNAPSHOT"
```
Comment (markfisher) [inline initializr-service/application.yml]: version is 1.0.0...

File: initializr-service/application.yml [hunk 4]
```diff
@@ -441,6 +441,16 @@ initializr:
...
+      content:
+        - name: Local Dataflow Server
+          id: cloud-dataflow-server-local
+          description: Local Dataflow Server implementation
```
Comment (markfisher) [inline initializr-service/application.yml]: 2 words: Data Flow

**Comment by snicoll** [comment]
I am curious, is that the only "starter" from the Spring Cloud Data Flow project? Spring Initializr can also add regular libraries if that makes sense. As long as the cloud bom provides dependency management for it, I am happy.

**Comment by dsyer** [comment]
The cloud be does not provide dependency management for data flow. I guess we need an extra BOM?

**Comment by joshlong** [comment]
this [https://github.com/spring-cloud/spring-cloud-starters/pull/42](PR) adds the version for the individual library to the imported dependencies project which would mean that the Initializr generated project would do the right thing and have a qualified version. we could switch it out to defer version management to a BOM if that's ultimately created.

**Comment by snicoll** [comment]
I am really not keen on adding more complexity at this point. When is it scheduled that data-flow joins the cloud release train?

**Comment by snicoll** [comment]
@joshlong actually, no. Have you tried your PR? A fresh build adding that new dependency gives me this

```
[ERROR]   The project com.example:demo:0.0.1-SNAPSHOT (/Users/snicoll/Downloads/demo 10/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for org.springframework.cloud:spring-cloud-starter-dataflow-server-local:jar is missing. @ line 27, column 15
```

which isn't surprising since no version is provided.

**Comment by joshlong** [comment]
I redid the PR in light of the newly available BOM for Spring Cloud Dataflow (see #189)

**Pull request closed without merging**